accounts/views.py

```diff
@@ -210,3 +214,129 @@ class ProfileDetailAPIView(RetrieveAPIView):
+        except User.DoesNotExist:
+            return Response(
+                {"error": "User not found"},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+        except Profile.DoesNotExist:
+            return Response(
+                {"error": "Profile not found"},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        # Check if already following
+        if current_profile.is_following(target_profile):
+            return Response(
+                {"error": "Already following this user"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        # Prevent self-follow (additional check, though model constraint handles this)
+        if current_profile == target_profile:
+            return Response(
+                {"error": "Cannot follow yourself"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        # Follow the user
+        current_profile.follow(target_profile)
+
+        serializer = FollowActionSerializer(
+            {
+                "message": "Successfully followed user",
+                "is_following": True,
+                "followers_count": target_profile.get_followers_count(),
+            },
+        )
+
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    def delete(self, request, user_id):
+        """Unfollow a user"""
+        try:
+            target_user = User.objects.get(id=user_id)
+            target_profile = Profile.objects.get(user=target_user)
+            current_profile = Profile.objects.get(user=request.user)
+        except User.DoesNotExist:
+            return Response(
+                {"error": "User not found"},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+        except Profile.DoesNotExist:
+            return Response(
+                {"error": "Profile not found"},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        # Check if not following
+        if not current_profile.is_following(target_profile):
+            return Response(
+                {"error": "Not following this user"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        # Unfollow the user
+        current_profile.unfollow(target_profile)
+
+        serializer = FollowActionSerializer(
+            {
+                "message": "Successfully unfollowed user",
+                "is_following": False,
+                "followers_count": target_profile.get_followers_count(),
+            },
+        )
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class UserFollowingListView(ListAPIView):
+    authentication_classes = [JWTAuthentication]
+    permission_classes = [IsAuthenticated]
+    serializer_class = ProfileFollowingSerializer
+    pagination_class = PageNumberPagination
+
+    def get_queryset(self):
+        user_id = self.kwargs["user_id"]
```

Comment:
`self.kwargs["user_id"]` have possibility to return error since we trying to access the key directly.

Better to use `get` function

```python
user_id = self.kwargs.get("user_id", None)
    if user_id is None:
        raise ValueError("user_id not found in URL kwargs.")
```